tests/backup_helper.go

```diff
@@ -10471,3 +10493,341 @@ func IsPxInstalled() bool {
+	backupDriver := Inst().Backup
+	shareClusterRequest := &api.ShareClusterRequest{
+		OrgId: BackupOrgID,
+		ClusterRef: &api.ObjectRef{
+			Name: clusterName,
+			Uid:  clusterUid,
+		},
+		// userid of the user(s) to share the cluster with
+		Users: users,
+		// group(s) to share the cluster with
+		Groups: groups,
+		// share_cluster_backups share is optional, if set to true, it will additionally share existing backups
+		ShareClusterBackups: shareClusterBackups,
+	}
+	resp, err := backupDriver.ShareCluster(ctx, shareClusterRequest)
+	if err != nil {
+		return resp, err
+	}
+	// Validate the share cluster
+	err = ValidateShareCluster(ctx, clusterName, clusterUid, users, groups)
+	if err != nil {
+		return nil, err
+	}
+
+	if shareClusterBackups {
+		err = ValidateShareClusterBackup(ctx, clusterName, clusterUid, users, groups)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return resp, nil
+}
+
+// UnShareCluster unshares a cluster with users and groups.
+func UnShareCluster(ctx context1.Context, clusterName string, clusterUid string, users []string, groups []string) (*api.UnShareClusterResponse, error) {
+	backupDriver := Inst().Backup
+	unshareClusterRequest := &api.UnShareClusterRequest{
+		OrgId: BackupOrgID,
+		ClusterRef: &api.ObjectRef{
+			Name: clusterName,
+			Uid:  clusterUid,
+		},
+		// userid of the user(s) to share the cluster with
+		Users: users,
+		// group(s) to share the cluster with
+		Groups: groups,
+	}
+	resp, err := backupDriver.UnShareCluster(ctx, unshareClusterRequest)
+	if err != nil {
+		return resp, err
+	}
+	return resp, nil
+}
+
+// UnShareClusterWithValidation unshares a cluster with users and groups and validates the unshare.
+func UnShareClusterWithValidation(ctx context1.Context, clusterName string, clusterUid string, users []string, groups []string) (*api.UnShareClusterResponse, error) {
+	backupDriver := Inst().Backup
+	unshareClusterRequest := &api.UnShareClusterRequest{
+		OrgId: BackupOrgID,
+		ClusterRef: &api.ObjectRef{
+			Name: clusterName,
+			Uid:  clusterUid,
+		},
+		// userid of the user(s) to share the cluster with
+		Users: users,
+		// group(s) to share the cluster with
+		Groups: groups,
+	}
+	resp, err := backupDriver.UnShareCluster(ctx, unshareClusterRequest)
+	if err != nil {
+		return resp, err
+	}
+
+	err = ValidateUnShareCluster(ctx, clusterName, clusterUid, users, groups)
+	if err != nil {
+		return nil, err
+	}
+	return resp, nil
+}
+
+// ValidateShareCluster validates that a cluster is shared with users and groups by validating collaborators list and inspecting cluster from user.
+func ValidateShareCluster(ctx context1.Context, clusterName string, clusterUid string, users []string, groups []string) error {
+	backupDriver := Inst().Backup
+	userIds := []string{}
+	groupIds := []string{}
+	req := &api.ClusterInspectRequest{
+		Name:  clusterName,
+		OrgId: BackupOrgID,
+		Uid:   clusterUid,
+	}
+
+	// Inspect cluster to retrieve collaborators
```

Comment:
These are functional level validation and more specificly this seems to me like a unit test case. than system test. Its good to have but will add test run time.

I would suggest to validate access through Backup Operations such as
Create Backup by shared user on the cluster
Create backup by user part of the group to which the cluster is shared.

Reply:
These are part of the helper method validation. All other backup-related CRUD operations will be validated at the test case level. This serves as a basic validation for the system tests.